### DISCUSSION TITLE
Help with starting off..

### DISCUSSION THREAD

**Discussion by dabailey2009** [ExperienceLovelace/ha-floorplan#102] [Q&A]
So I am very very nooby...  Sorry.  I am having trouble getting started with floorplan.  The aim is to take .dwg files and convert them as a basis for the floorplan and potentially also some 3D renders later on, but let's not get ahead of ourselves...
I come to the following 
>"Remember to add the /hacsfiles/ha-floorplan/floorplan.js as a module to your resource file - like you do with other modules."
This is the first module I have done so "like you do with other modules" is a bit cryptic.
I go to configuration but it just takes me back to my sandbox lovelace ui

![Configuration_Screenshot_2021-05-22_10-53-00](https://user-images.githubusercontent.com/79867421/119220675-fbd42f80-baeb-11eb-84e4-80dfc14363df.png)
![Screenshot_2021-05-22_10-54-26](https://user-images.githubusercontent.com/79867421/119220711-20c8a280-baec-11eb-8acd-61da6ec87539.png)

I have tried adding a new card and finding various yaml templates that are in Github like this one but with now joy... Help gratefully received!
https://github.com/ExperienceLovelace/ha-floorplan/blob/f94b4f18b44e29ba01e1f2a8074146355bdc7b0d/docs/_docs/floorplan/examples/home/home.lovelace-view.yaml

**Comment by exetico** [comment]
Hi,

Have you followed the https://experiencelovelace.github.io/ha-floorplan/docs/quick-start/ and continued with https://community.home-assistant.io/t/floorplan-now-available-as-a-lovelace-card/115489/323?u=exetico?

You'll need a SVG file. And I'll recommend you to try one of the examples first, it you're unsure there to start. 

After **that's done**, you can try and use your own SVG file 😊👍.

**Reply by dabailey2009** [reply]
Thanks for getting back so quickly.  Yes.  This is the bit I am stuck with.

"Remember to add the /hacsfiles/ha-floorplan/floorplan.js as a module to your resource file - like you do with other modules."

**Reply by dabailey2009** [reply]
If I paste this into the custom card, I just end up with card filled with code.
![Screenshot_2021-05-22_11-36-40](https://user-images.githubusercontent.com/79867421/119222011-7e5fed80-baf2-11eb-8ab4-843ec0776cee.png)

**Reply by OzGav** [reply]
Have a look at this discussion as I think it will answer some of your questions https://github.com/ExperienceLovelace/ha-floorplan/discussions/97

**Reply by dabailey2009** [reply]
Thanks @OzGav I am running the homeassistant on the pi and ssh ing into it so there are a few basic things that I am probably cocking up.  I will muck about with it a bit more - this post is certainly helpful but I still got lost and eventually gave up.

**Reply by OzGav** [reply]
You don't need to SSH in to do this. You can set up the SAMBA share addon and then just edit the ui-lovelace.yaml with your favourite text editor. OR you can just do the whole thing via the UI. I use the samba share (On a Pi) and edit with Notepad++

If you go to CONFIGURATION>>LOVELACE DASHBOARDS>>RESOURCES do you see an entry for `floorplan.js`  ?

**Comment by dabailey2009** [comment]
So in configuration.yaml I only have 

# Configure a default setup of Home Assistant (frontend, api, etc)
default_config:

# Text to speech
tts:
  - platform: google_translate

group: !include groups.yaml
automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml

But I need to get back to this a bit later -  Thanks very much for you help I am definitely just doing something daft

**Reply by OzGav** [reply]
No not configuration.yaml. In your Home Assistant interface you have the series of icons on the left side. Down the bottom there is a GEAR that is the CONFIGURATION page. From there find LOVELACE DASHBOARDS then when you click on that you should find two tabs twoards the top. Click on RESOURCES. If floorplan.js isn't there then you need to add it with the button bottom right. You need to know the path to your floorplan.js file. If you used HACS it is `/hacsfiles/ha-floorplan/floorplan.js` the type is javascript module

**Reply by dabailey2009** [reply]
Thanks - but It seems like I just don't get the two tabs at the top...
![image](https://user-images.githubusercontent.com/79867421/119838891-0fc0cc80-bf04-11eb-911e-ed56d81cd14b.png)

**Reply by OzGav** [reply]
OK that's weird. I am not sure what HA installation you are using. You can try and add it manually by putting this in your configuration.yaml

```
lovelace:
  mode: yaml
  # Include external resources only add when mode is yaml, otherwise manage in the resources in the lovelace configuration panel.
  resources:
  - type: module
    url: /hacsfiles/ha-floorplan/floorplan.js
```
